bin/hdfcoinc/pycbc_fit_sngls_over_param

```diff
@@ -0,0 +1,177 @@
+elif args.fit_param in pnutils.named_frequency_cutoffs.keys():
+    parvals = pnutils.frequency_cutoff_from_name(args.fit_param, m1, m2, s1z, s2z)
+elif args.fit_param == 'template_duration':
+    parvals = pnutils.get_seobnrrom_duration(m1, m2, s1z, s2z, f_low=args.f_lower)
+    parvals += args.min_duration
+else:
+    # try asking for a LALSimulation frequency function
+    parvals = pnutils.get_freq(args.fit_param, m1, m2, s1z, s2z)
+
+nabove = fits['count_above_thresh'][:]
+nabove = nabove/np.mean(nabove)
+# for an exponential fit 1/alpha is linear in the trigger statistic values
+# so taking weighted sums/averages of 1/alpha is appropriate
+invalpha = 1./(fits['fit_coeff'][:])
+
+# sort in ascending duration order
+parsort = np.argsort(parvals)
+tid = tid[parsort]
+parvals = parvals[parsort]
+nabove = nabove[parsort]
+invalpha = invalpha[parsort]
+
+if args.log_param:
+    logging.info('Using log %s to perform smoothing' % args.fit_param)
+    parvals = np.log(parvals)
+else:
+    logging.info('Using %s to perform smoothing' % args.fit_param)
+
+# do nearest-neighbours regression
+# use Gaussian weight over fitting parameter
+if args.regression_method == 'nn':
+    weights = lambda d:np.exp(-0.5 * (d/args.smoothing_width)**2)
+    knn = neighbors.KNeighborsRegressor(args.num_neighbors, weights=weights)
+
+    logging.info('Smoothing nabove data')
+    nabove_knn = knn.fit(parvals[:, np.newaxis], nabove)
+    logging.info('Evaluating smoothed nabove')
+    nabove_smoothed = [nabove_knn.predict([[p]]) for p in parvals]
+    del nabove_knn
+
+    logging.info('Smoothing invalpha data')
+    # smooth the inverse alpha values times trig count above threshold
+    invalphan = invalpha * nabove
+    invalphan_knn = knn.fit(parvals[:, np.newaxis], invalphan)
+    logging.info('Evaluating smoothed invalpha')
+    # loop over parameter values to avoid memory error in knn predict
+    invalphan_smoothed = [invalphan_knn.predict([[p]]) for p in parvals]
+    del invalphan_knn
+    # divide out by smoothed trig count
+    invalpha_smoothed = np.array(invalphan_smoothed) / np.array(nabove_smoothed)
+
+# store template-dependent fit output
+outfile = h5py.File(args.output, 'w')
+outfile.create_dataset('template_id', data=tid)
+outfile.create_dataset('count_above_thresh', data=nabove_smoothed)
+outfile.create_dataset('fit_coeff', data=1. / np.array(invalpha_smoothed))
+
+# add metadata, some is inherited from template level fit
+outfile.attrs.create('ifo', data=ifo)
+outfile.attrs.create('stat_threshold', data=fits.attrs['stat_threshold'])
+outfile.attrs.create('fit_param', data=args.fit_param)
+outfile.attrs.create('regression_method', data=args.regression_method)
+if args.num_neighbors > 0:
+    outfile.attrs.create('n_neighbors', data=args.num_neighbors)
+outfile.attrs.create('smoothing_width', data=args.smoothing_width)
+
+# add a magic file attribute so that coinc_findtrigs can parse it
+outfile.attrs.create('stat', data=ifo+'-fit_coeffs')
+
```

Comment:
You don't need to change this, but for reference, I would have thought the following looks cleaner.
```
outfile.attrs['ifo'] = ifo
outfile.attrs['stat_threshold'] = fit.attrs['stat_threshold']
...and so on...
```